SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/LayeredLBStrategy/LayeredLBStrategyTest.cpp

```diff
@@ -0,0 +1,430 @@
+	ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategyLocalWorker(Data, 2));
+
+	this->AddExpectedError("The Local Virtual Worker Id cannot be set twice. Current value:",
+		EAutomationExpectedErrorFlags::MatchType::Contains, 1);
+
+	return true;
+}
+
+LAYEREDLBSTRATEGY_TEST(Given_two_actors_of_same_type_at_same_position_WHEN_who_should_have_auth_called_THEN_return_same_for_both)
+{
+	AutomationOpenMap("/Engine/Maps/Entry");
+
+	TSharedPtr<TestData> Data = TSharedPtr<TestData>(new TestData);
+
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld(Data));
+	ADD_LATENT_AUTOMATION_COMMAND(FCreateStrategy(Data));
+	ADD_LATENT_AUTOMATION_COMMAND(FSetDefaultLayer(Data, UGridBasedLBStrategy::StaticClass()));
+	ADD_LATENT_AUTOMATION_COMMAND(FAddLayer(Data, { UTwoByFourLBGridStrategy::StaticClass(), {ALayer1Pawn::StaticClass()} }));
+	ADD_LATENT_AUTOMATION_COMMAND(FAddLayer(Data, { UTwoByFourLBGridStrategy::StaticClass(), {ALayer2Pawn::StaticClass()} }));
+	ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy(Data, {}));
+	ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategyLocalWorker(Data, 1));
+	ADD_LATENT_AUTOMATION_COMMAND(FSpawnLayer1PawnAtLocation(Data, TEXT("Layer1Actor1"), FVector(0, 0, 0)));
+	ADD_LATENT_AUTOMATION_COMMAND(FSpawnLayer1PawnAtLocation(Data, TEXT("Layer1Actor2"), FVector(0, 0, 0)));
+	ADD_LATENT_AUTOMATION_COMMAND(FSpawnLayer2PawnAtLocation(Data, TEXT("Layer2Actor1"), FVector(0, 0, 0)));
+	ADD_LATENT_AUTOMATION_COMMAND(FSpawnLayer2PawnAtLocation(Data, TEXT("Later2Actor2"), FVector(0, 0, 0)));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FCheckActorsAuth(Data, this, TEXT("Layer1Actor1"), TEXT("Layer1Actor2"), true));
+	ADD_LATENT_AUTOMATION_COMMAND(FCheckActorsAuth(Data, this, TEXT("Layer2Actor1"), TEXT("Later2Actor2"), true));
+
+	return true;
+}
+
+LAYEREDLBSTRATEGY_TEST(GIVEN_two_actors_of_different_types_and_same_positions_managed_by_different_layers_WHEN_who_has_auth_called_THEN_return_different_values)
+{
+	AutomationOpenMap("/Engine/Maps/Entry");
+
+	TSharedPtr<TestData> Data = TSharedPtr<TestData>(new TestData);
+
+	ADD_LATENT_AUTOMATION_COMMAND(FWaitForWorld(Data));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FCreateStrategy(Data));
+	ADD_LATENT_AUTOMATION_COMMAND(FSetDefaultLayer(Data, UGridBasedLBStrategy::StaticClass()));
+	ADD_LATENT_AUTOMATION_COMMAND(FAddLayer(Data, { UTwoByFourLBGridStrategy::StaticClass(), {ALayer1Pawn::StaticClass()} }));
+	ADD_LATENT_AUTOMATION_COMMAND(FAddLayer(Data, { UTwoByFourLBGridStrategy::StaticClass(), {ALayer2Pawn::StaticClass()} }));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategy(Data, {}));
+	ADD_LATENT_AUTOMATION_COMMAND(FSetupStrategyLocalWorker(Data, 1));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FSpawnLayer1PawnAtLocation(Data, TEXT("Layer1Actor"), FVector(0, 0, 0)));
+	ADD_LATENT_AUTOMATION_COMMAND(FSpawnLayer2PawnAtLocation(Data, TEXT("Layer2Actor"), FVector(0, 0, 0)));
+
+	ADD_LATENT_AUTOMATION_COMMAND(FCheckActorsAuth(Data, this, TEXT("Layer1Actor"), TEXT("Layer2Actor"), false));
+
+	return true;
+}
+}
```

Comment:
```suggestion
} // anonymous namespace
```
also in other test files we close the namespace just after our last function definition without unreal macro (so in this case I think it would be `GetAnyGameWorld`)

Reply:
moved namespace +  added comment